### PR TITLE
is_obstructed? method with tests

### PR DESCRIPTION
**Pull request by kellylougheed**
Added piece.is_obstructed? method to check for horizontal, vertical, and diagonal obstruction. It does not check if the destination is occupied or if the move is valid, just for obstructions. Sixteen tests check for true and false results in eight directions.

### CODE REVIEW

**Comment by lasley** [comment]
If ready for review, drop a ready for review tag. It's also typically good to drop a comment too so everyone watching the repo gets pinged.

**Comment by kellylougheed** [comment]
Let me know what you guys think!

File: app/models/piece.rb [hunk 1]
```diff
@@ -2,5 +2,47 @@ class Piece < ActiveRecord::Base
...
+
+  def is_obstructed?(destination_x, destination_y)
+    location_x = self.x_position
+    location_y = self.y_position
+    if location_x == destination_x
+      # check for vertical obstruction
+      location_y > destination_y ? incrementer = -1 : incrementer = 1
+      position_y = location_y + incrementer
+      while position_y != destination_y
+        if game.pieces.where(x_position: location_x, y_position: position_y).any?
```
Comment (lasley) [inline app/models/piece.rb]: Maybe possible to simplify by creating a range of your `position_y..destination_y`, then doing a `where(x_position: location_x, y_position: my_range)`

Comment (kellylougheed) [inline app/models/piece.rb]: @lasley I like the idea and played around with it, but there are complications with reverse ranges. When the piece is moving down or to the left, you get a reverse range like 4...1 and it seems like Ruby doesn't like those: http://stackoverflow.com/questions/2070574/is-there-a-reason-that-we-cannot-iterate-on-reverse-range-in-ruby

Comment (lasley) [inline app/models/piece.rb]: The problem is that we need to just order which one goes first in the range. What about `[y, x].minmax`

File: app/models/piece.rb [hunk 2]
```diff
@@ -2,5 +2,41 @@ class Piece < ActiveRecord::Base
...
   belongs_to :player
+
+  def is_obstructed?(destination_x, destination_y)
+    location_x = self.x_position
+    location_y = self.y_position
+    if location_x == destination_x
+      # check for vertical obstruction
+      location_y > destination_y ? incrementer = -1 : incrementer = 1
+      position_y = location_y + incrementer
+      range = [position_y, destination_y].min...[position_y, destination_y].max
```
Comment (lasley) [inline app/models/piece.rb]: there's actually a .minmax function that would be slightly more efficient, but this works too. 
```
min, max = [position_y, destination_y].minmax
range = min...max
```

**Pull request merged**